### PR TITLE
Extension cases

### PR DESCRIPTION
**Pull request by kaapstorm**
Not a "PR" PR. Just opening this for discussion. 

I've added a few test stubs, and I've chosen to implement this using a new FormAction. I think that will keep things cleaner than adding to existing subcases code to accommodate extensions. Feel free to suggest more tests. And shout if you think I'm taking the wrong approach with adding OpenExtCaseAction. There's some WIP I've stashed, and I'll push that as it progresses.

@ctsims, @orangejenny, @sheelio, cc @snopoke, @biyeun

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_extension_case.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        form.get_ext_case_types should return ext case types
+        """
+
+    @skip
+    def test_module_get_ext_case_types(self):
+        """
+        module.get_ext_case_types should return ext case types
+        """
+
+class ExtCaseAdvancedFormCaseXmlTests(SimpleTestCase):
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_extension_case.py]: E302 expected 2 blank lines, found 1

**Comment by snopoke** [comment]
I was thinking about this on the way home. Should we not just consider the
subcase action to be 'index-case-action'. If we do that then we can just
add a property to distinguish child from extension.

I haven't fully thought this through so could be that it doesn't make sense
but I think it should work. Let's chat tomorrow.

**Comment by kaapstorm** [comment]
That's exactly the alternative I was thinking of; an `is_extension` flag. I guess what will tip the balance one way or the other is whether extension cases are similar enough to child cases to use a flag, or different enough to use an action.

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/apps/app_manager/tests/test_extension_case.py, line 5 - F401 'XFormCaseBlock' imported but unused
* corehq/apps/app_manager/tests/test_extension_case.py, line 80 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 81 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 106 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 107 - E131 continuation line unaligned for hanging indent

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/apps/app_manager/tests/test_extension_case.py, line 5 - F401 'XFormCaseBlock' imported but unused
* corehq/apps/app_manager/tests/test_extension_case.py, line 80 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 81 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 106 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 107 - E131 continuation line unaligned for hanging indent

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/apps/app_manager/tests/test_extension_case.py, line 5 - F401 'XFormCaseBlock' imported but unused
* corehq/apps/app_manager/tests/test_extension_case.py, line 80 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 81 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 106 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 107 - E131 continuation line unaligned for hanging indent

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/app_manager/tests/test_extension_case.py, line 3 - F401 're' imported but unused
* corehq/apps/app_manager/tests/test_extension_case.py, line 9 - F401 'XFormCaseBlock' imported but unused
* corehq/apps/app_manager/tests/test_extension_case.py, line 143 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 144 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 169 - E131 continuation line unaligned for hanging indent
* corehq/apps/app_manager/tests/test_extension_case.py, line 170 - E131 continuation line unaligned for hanging indent

**Pull request closed without merging**